main.py

```diff
@@ -313,6 +315,100 @@ async def get_running_experiment() -> Optional[int]:
+    with open(metadata_path, encoding="utf-8") as metadata_file:
+        metadata = json.load(metadata_file)
+    submission_time_str, visualize = metadata["submission_time"], metadata["visualize"]
+    submission_time = datetime.fromisoformat(submission_time_str)
+    date = submission_time.date().isoformat()
+    hour = submission_time.hour
+    # find the result file
+    datetime_result_dir_path = posixpath.join(result_dir_path, f"{date}/{hour}/")
+    padded_rid = rid.zfill(9)
+    for item in os.listdir(datetime_result_dir_path):
+        if item.startswith(padded_rid):
+            break
+    else:  # no result file
+        return False
+    # copy the result file to the RID directory
+    result_path = posixpath.join(datetime_result_dir_path, item)
+    copied_result_path = posixpath.join(rid_dir_path, "result.h5")
+    shutil.copyfile(result_path, copied_result_path)
+    # modify the result file
+    with h5py.File(copied_result_path, "a") as result_file:
+        submission_time_dataset = result_file.create_dataset(
+            "submission_time", (1,), dtype=h5py.string_dtype(encoding="utf-8")
+        )
+        submission_time_dataset[0] = submission_time_str
+        visualize_dataset = result_file.create_dataset("visualize", (1,), dtype="bool")
+        visualize_dataset[0] = visualize
+    # move the modified experiment file to the RID directory
+    if visualize:
+        experiment_path = posixpath.join(result_dir_path, f"experiment_{submission_time_str}.py")
+        moved_experiment_path = posixpath.join(rid_dir_path, "modified_experiment.py")
+        shutil.move(experiment_path, moved_experiment_path)
+    # remove the metadata file
+    os.remove(metadata_path)
+    return True
+
+
+@app.get("/result/")
+async def list_result_directory() -> List[int]:
+    """Post-processes the submitted experiments and returns the RID list.
+
+    It performs the following:
+        1. Organize the result directory.
+        2. Return the RID list.
+
+    Returns:
+        A list with RIDs of the submitted experiments, sorted in an ascending order.
+    """
+    rid_list = []
+    # read the most recently fetched RID
+    result_dir_path = posixpath.join(configs["master_path"], configs["result_path"])
+    last_rid_path = posixpath.join(result_dir_path, "rid.json")
+    try:
+        with open(last_rid_path, encoding="utf-8") as last_rid_file:
+            last_rid = json.load(last_rid_file)
+    except FileNotFoundError:
+        last_rid = -1
+    # navigate to each RID directory
+    for item in os.listdir(result_dir_path):
+        item_path = posixpath.join(result_dir_path, item)
+        if posixpath.isdir(item_path) and item.isdigit():  # find a RID directory
+            rid = int(item)
+            if rid <= last_rid:  # alraedy organized
+                rid_list.append(rid)
+                continue
+            if organize_result_directory(result_dir_path, item):
+                rid_list.append(rid)
```

Comment:
Is it common to have no result file? Don't we need to notify it, e.g., by logging?

Reply:
It may occur commonly!

For example, you submit a long-time experiment and fetch the RID list right after.
When the experiment is submitted, the RID directory is created, but it will take a long time to create the result file.
In this case, it must not organize this RID directory until it finishes.

In summary, a running or queued experiment must not be organized!